### PR TITLE
Always create MapReduce job jar on local fs.

### PR DESCRIPTION
**Pull request by chtyim**
- Job.submit will copy it to HDFS
- Need this in order to work on both local, HDFS and MapR FS

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java [hunk 1]
```diff
@@ -699,12 +702,18 @@ private Location buildJobJar(BasicMapReduceContext context) throws IOException {
...
-    Location jobJar =
-      locationFactory.create(String.format("%s.%s.%s.%s.%s.jar",
-                                           ProgramType.MAPREDUCE.name().toLowerCase(),
-                                           programId.getNamespaceId(), programId.getApplicationId(),
-                                           programId.getId(), context.getRunId().getId()));
-
+    // Build the jobJar on local temp directory. On Job.submit(), the MapReduce framework will copy the file to HDFS
+    // and distribute it to task containers.
+    // It has to be local, otherwise it won't work on MapR.
+    File tempDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
```
Comment (jwang47) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java]: Where or when is this cleaned up?

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java]: tempDir is the "data/tmp" directory, which the directory itself won't get delete (in standalone, the content will get removed on start).

In distributed mode, it's the container directory, hence also don't need to worry about the cleanup. The JobJar file created inside the tempDir already get cleanup properly (existing code, cleanup on job finish as well as any exception happens in between).

**Comment by gsps1** [comment]
LGTM :+1:

**Comment by chtyim** [comment]
Verified in all different mode.

**Pull request merged**